### PR TITLE
Cleanup state declarations.

### PR DESCRIPTION
**Pull request by pavanbalaji**

### CODE REVIEW

**Comment by pavanbalaji** [comment]
test:jenkins/ch4/all

File: src/mpid/ch4/src/ch4_spawn.h [hunk 1]
```diff
@@ -40,17 +43,23 @@ static inline int MPIDI_mpi_to_pmi_keyvals(MPIR_Info * info_ptr,
...
 
-  fn_fail:
   fn_exit:
     *kv_ptr = kv;
     *nkeys_ptr = nkeys;
+    MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIDI_MPI_TO_PMI_KEYVALS);
     return mpi_errno;
+
+  fn_fail:
+    goto fn_exit;
```
Comment (wesbland) [inline src/mpid/ch4/src/ch4_spawn.h]: Technically, this could be a separate commit. I don't care that much though.

Comment (pavanbalaji) [inline src/mpid/ch4/src/ch4_spawn.h]: Yeah, I'll split it out and signoff on your behalf.
